src/controllers/tasks.controller.ts

```diff
@@ -1,19 +1,25 @@
-import { Task, TaskStatus } from '../interface/task.type';
+import { z } from 'zod';
+import { TaskStatus, taskSchema } from '../model/zod.schema';
 import { v4 as uuidv4 } from 'uuid';
 import { filePath } from '../utils/checkFile';
 import { writeJsonFile, readJsonFile } from '../utils/jsonOperations';
 
-export const createTask = async (options: Omit<Task, 'status' | 'isArchived'>): Promise<void> => {
-  const newTask: Task = {
-    id: uuidv4(),
-    name: options.name,
-    deadline: new Date(options.deadline),
-    status: TaskStatus.TODO,
-    tags: options.tags || [],
-    isArchived: false,
-  };
-  let tasks = [];
+export const createTask = async (options: z.infer<Omit<typeof taskSchema, 'status' | 'isArchived'>>): Promise<void> => {
   try {
+    const newTask: z.infer<typeof taskSchema> = {
+      id: uuidv4(),
+      name: options.name,
+      deadline: new Date(options.deadline),
+      status: TaskStatus.TODO,
+      tags: options.tags,
+      isArchived: false,
+    };
+    let tasks = [];
```

Comment:
```suggestion
    let tasks: Task[] = [];
```
When you intialize an array without any type, it becomes `any[]`